Entities/Industry/Drill/Drill.as

```diff
@@ -388,9 +391,44 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
 void onAttach(CBlob@ this, CBlob@ attached, AttachmentPoint @attachedPoint)
 {
 	this.getCurrentScript().runFlags &= ~Script::tick_not_sleeping;
+	this.set_s32("showHeatTo", getPlayerIndex(attached.getPlayer()));
```

Comment:
With the current implementation, I assume a random player will be shown the heat indicator if that player leaves. You should instead be storing the player's network ID:

`this.set_u16("showHeatTo", attached.getPlayer().getNetworkID());`

Reply:
I have changed the code to store the player's network ID instead of the player index